### PR TITLE
ENH: add reading code which returns unique uids from qr via zbarcam

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Need to add xonsh to deps

File: sampledb/sample_schema.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "type" : "string",
+    "required" : true,
+    "description" : ""
+ },
+ "phase_info": {
+    "type" : ["string", "array"],
+    "required" : true,
+    "description" : ""
+ },
+ "sample-name_of_sample_background": {
```
Comment (CJ-Wright) [inline sampledb/sample_schema.json]: Might default this to air?

File: sampledb/spreadsheet.xsh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    writer = pd.ExcelWriter('test.xlsx', engine='xlsxwriter')
+    df.to_excel(writer, index=False)
+    sheet = writer.sheets['Sheet1']
+    for i, name in enumerate(df):
+        width = max(len(str(val)) for val in df[name])
+        width = max(width, len(name)) + 1
+        sheet.set_column(i, i, width)
+    writer.save()
+    
+    soffice 'test.xlsx'
```
Comment (CJ-Wright) [inline sampledb/spreadsheet.xsh]: maybe make the file name configurable

File: sampledb/sample_schema.json (hunk 1 above)
Comment (justcalamari) [reply sampledb/sample_schema.json]: Sounds like a good idea.

File: sampledb/spreadsheet.xsh (hunk 2 above)
Comment (justcalamari) [reply sampledb/spreadsheet.xsh]: I thought we would remove the file after the code finishes running, so I'm not sure how useful a configurable file name would be. Though "test.xlsx" probably isn't the best name.

Comment (justcalamari) [reply sampledb/spreadsheet.xsh]: We obviously don't want to overwrite and remove another file a user has with the same name, so maybe we can check if the file exists beforehand, and if it does the code can instead use "test-1.xlsx".

Comment (CJ-Wright) [reply sampledb/spreadsheet.xsh]: Do we care about the names? If not just make them the timestamp (they are usually unique, and convey some info).

Comment (justcalamari) [reply sampledb/spreadsheet.xsh]: That works.

File: sampledb/reader.xsh [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+from time import time
+from sampledb.sampledatabase import SampleDatabase
+
+
+def get_uid_from_qr():
+    a = $(zbarcam)
+    b = a.split()
+    c = [z.strip('QR-Code:') for z in b]
+    d = set(c)
```
Comment (CJ-Wright) [inline sampledb/reader.xsh]: We should use an OrderedDict here so that we preserve the order of the unique elements.

File: sampledb/reader.xsh [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    return d
+
+def write_sample_spreadsheet(uids, sdb):
+    if len(uids) == 0:
+        return
+
+    sdb.load_schema('sample_schema.json')
```
Comment (CJ-Wright) [inline sampledb/reader.xsh]: We might want a dedicated order to the fields in excel, at least for consistency.

File: sampledb/sample_schema.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
+    "$schema": "http://json-schema.org/draft-06/schema#",
+    "title": "Sample",
+    "type": "object",
+    "required": ["uid", "sample_name", "phase_info",
+        "sample-name_of_sample_background", "sample_maker"],
```
Comment (CJ-Wright) [inline sampledb/sample_schema.json]: add project

**Pull request merged**